spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -244,15 +244,18 @@ class Recurrent[T : ClassTag]()
      * identical elements T(output, output). One of the elements from the cell output is
      * the updated hidden. Thus the currentInput will update its hidden element with this output.
      */
-    currentInput(hidDim) = hidden
     var i = 1
+    // init state
+    currentInput(hidDim) = if (initState != null) initState
+     else hidden
     while (i <= times) {
       currentInput(inputDim) = outputCell.select(timeDim, i)
       cells(i - 1).updateOutput(currentInput)
       currentInput(hidDim) = cells(i - 1).output.toTable(hidDim)
       i += 1
     }
 
+    initState = null
```

Comment:
As synced offline, It's expected behavior, if want to use state from previous layer, we need explicitly call setState(). I am not very like to propogate state in Recurrent, the code will not easy to maintain and less readable. Leave propogate state in up level layer like other framework do.